Formula/watchexec.rb

```diff
@@ -0,0 +1,18 @@
+class Watchexec < Formula
+  desc "Execute commands when watched files change"
+  homepage "https://github.com/mattgreen/watchexec"
+  url "https://github.com/mattgreen/watchexec/archive/1.6.0.tar.gz"
+  sha256 "0ce7b26bbc05400a835b5a06e13ef69aac90a6d93ef4b989241258494f3c967f"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "build", "--release"
+    bin.install "target/release/watchexec"
+    man1.install "doc/watchexec.1"
+  end
+
+  test do
+    system "#{bin}/watchexec", "--version"
```

Comment:
It is difficult to make a more complete test since `watchexec` works by monitoring for filechange events and then executing a command. It does not exit after being called.

I tried the following.

```diff
diff --git a/Formula/watchexec.rb b/Formula/watchexec.rb
index d6a59e5..ee3c0a0 100644
--- a/Formula/watchexec.rb
+++ b/Formula/watchexec.rb
@@ -13,6 +13,9 @@ class Watchexec < Formula
   end
 
   test do
-    system "#{bin}/watchexec", "--version"
+    o = IO.popen("#{bin}/watchexec -- echo 'saw file change'")
+    sleep 0.1
+    Process.kill("INT", o.pid)
+    assert_equal "saw file change\n", o.read
   end
 end
```

Which works but is not very pleasant. 

*Note:* I don't know ruby.

`watchexec` is pretty similar to [`watchman`][1], which also has a pretty rudimentary test.

[1]: https://github.com/Homebrew/homebrew-core/blob/7dfe465b7cb48be58bedc4349a5357953a53133e/Formula/watchman.rb#L47